packages/admin/cms-admin/src/common/MasterMenuRoutes.tsx

```diff
@@ -7,24 +7,45 @@ import { MasterMenuData, MasterMenuItem } from "./MasterMenu";
 
 export function useRoutePropsFromMasterMenuData(items: MasterMenuData): RouteProps[] {
     const isAllowed = useUserPermissionCheck();
-    const checkPermission = (item: MasterMenuItem): boolean => !item.requiredPermission || isAllowed(item.requiredPermission);
+    const checkPermission = (item: MasterMenuItem, ancestors: MasterMenuItem[]): boolean => {
+        if (item.requiredPermission) {
+            return isAllowed(item.requiredPermission);
+        }
+
+        for (const ancestor of ancestors) {
```

Comment:
Do I need to reverse the ancestors here?

Reply:
Changed the logic to consider the item and all ancestors: 1c2f34c6e266d5b434f6a601abe9aca1a53c54aa